### PR TITLE
Update load of search param status on initialization

### PR DESCRIPTION
**Pull request by feordin**
## Description
Sets the SearchParameterStatus property correctly upon initialization.  Also updates the DataActions to allow globalAdmin to run search parameter status changes.

## Related issues
Addresses [issue [AB#131939](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/131939)].

## Testing
Automated tests were rerun, manual testing done

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- Tag the PR with **Schema Version backward compatible** or **Schema Version backward incompatible** or **Schema Version unchanged** if this adds or updates Sql script which is/is not backward compatible with the code.
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Core/Features/Search/Registry/SearchParameterStatusManager.cs [hunk 1]
```diff
@@ -81,12 +81,14 @@ internal async Task EnsureInitializedAsync(CancellationToken cancellationToken)
                     if (p.IsSearchable != tempStatus.IsSearchable ||
                         p.IsSupported != tempStatus.IsSupported ||
                         p.IsPartiallySupported != tempStatus.IsPartiallySupported ||
-                        p.SortStatus != result.SortStatus)
+                        p.SortStatus != result.SortStatus ||
+                        p.SearchParameterStatus != result.Status)
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Core/Features/Search/Registry/SearchParameterStatusManager.cs:85]: Any way to add a test? https://github.com/microsoft/fhir-server/blob/323b9e8b7d40ad9fbc14f0fb0a54c5657d306416/src/Microsoft.Health.Fhir.Core.UnitTests/Features/Search/Registry/SearchParameterStatusManagerTests.cs

Comment (feordin) [reply src/Microsoft.Health.Fhir.Core/Features/Search/Registry/SearchParameterStatusManager.cs:85]: Updated a unit test to cover this case.

**Review by brendankowitz — APPROVED** [review]

**Review by mikaelweave — DISMISSED** [review]

**Review by mikaelweave — APPROVED** [review]

**Pull request merged**